# recA_analysis
Bioinformatic analysis of the bacterial gene recA to understand the role it plays in the evolution of antibiotic resistance

# Background
Based on clasical genetics, we hypothesize that recA is necessary for the evolution of antibiotic resistance.  recA is highly conserved across bacterial species and serves as a master regulator of the so-called "SOS Response", which is a series of genes turned on when bacteria experience stress, such as that induced by antibiotics.  Downstream genes include highly mutagenic polymerases, which allow a population to more rapidly evolve resistance genes.  Additionally, recA controls horizontal gene transfer, which allows a population to rapidly share such innovative resistance genes once a particular clone has "found" them.  This project uses bioinformatics tools to explore features the recA evolutionary process in E. coli.

# Part One: dN/dS evolutionary analysis


# Part Two: RNA-seq analysis
This segment of the project explores a recent data set generated by subjecting E. coli to 17 different antibiotics at sub-lethal concentrations for 25 minutes until OD ~0.2, and then performing RNA-seq [1].  This data is found on NCBI under accession number GSE110137.  We will use these data to perform the following steps:

1. Align the RNA reads to a reference sequence using the industry standard nextflow RNA-seq pipeline [2]. Note: first iteration aligned to the EB1 reference genome.  Future iterations will assemble the true reference genome from the actual strain of bacteria used in the study, BW25113, which is found under accession number CP009273 [3].  

2. Generate a network subgraph centered around recA and mutagenic polymerases.  We plan to use the differential expression matrix created in step 1 and we expect these genes to be highly expressed during the stress caused by antibiotics.  We will follow the proceedure detailed by Cabusora [4].  A similar approach for building a recA subgraph was employed in Mycobacterium tuberculosis [5]. 

3. Simulate a directed network attack against recA to evaluate recA as a therapeutic target.

# Running the code
For step 2.1, the workflow is run entirely using the nextflow RNA-seq pipeline with a docker profile from the command line, using the command listed in the file "RNA_seq_bash_commands".

To successfully implement this part of the pipeline, users first need to download an RNA-seq sample in FASTQ format.  This requires installing SRA tools from NCBI (https://github.com/ncbi/sra-tools).  These files are too large to be stored in this repo, but can be retreived using the bash command listed in the ./code directory on this repo.  

Next, these files are fed into the nextflow RNA-seq data pipeline, which performs sequence aligment using STAR and generates data QC reports at various stages of data processing.  This again requires setting up nextflow RNA-seq per the instructions at https://github.com/nf-core/rnaseq.  Once nextflow is set up with a docker profile, the pipeline can be run using the bash command listed in the ./code directory on this repo.  


# References
1. Zoffmann S, Vercruysse M, Benmansour F, et al. Machine learning-powered antibiotics phenotypic drug discovery. Sci Rep. 2019;9(1):5013. Published 2019 Mar 21. doi:10.1038/s41598-019-39387-9
2. Ewels PA, Peltzer A, Fillinger S, Alneberg JA, Patel H, Wilm A, Garcia MU, Di Tommaso P, Nahnsen S. nf-core: Community curated bioinformatics pipelines. bioRxiv. 2019. p. 610741. doi: 10.1101/610741 
3. Grenier F, Matteau D, Baby V, Rodrigue S. Complete Genome Sequence of Escherichia coli BW25113. Genome Announc. 2014;2(5):e01038-14. Published 2014 Oct 16. doi:10.1128/genomeA.01038-14
4. Lawrence Cabusora, Electra Sutton, Andy Fulmer, Christian V. Forst, Differential network expression during drug and stress response, Bioinformatics, Volume 21, Issue 12, , Pages 2898–2905, https://doi.org/10.1093/bioinformatics/bti440
5. Nicole Radde, Jutta Gebert, Christian V. Forst, Systematic component selection for gene-network refinement, Bioinformatics, Volume 22, Issue 21, 1 November 2006, Pages 2674–2680, https://doi.org/10.1093/bioinformatics/btl440
